datadog/resource_datadog_integration.go

```diff
@@ -0,0 +1,171 @@
+package datadog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+func resourceDatadogIntegrationPagerduty() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogIntegrationPagerdutyCreate,
+		Read:   resourceDatadogIntegrationPagerdutyRead,
+		Update: resourceDatadogIntegrationPagerdutyUpdate,
+		Delete: resourceDatadogIntegrationPagerdutyDelete,
+		Exists: resourceDatadogIntegrationPagerdutyExists,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogIntegrationPagerdutyImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"services": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "A list of service names and service keys.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"service_key": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"subdomain": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"schedules": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"api_token": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceDatadogIntegrationPagerdutyExists(d *schema.ResourceData, meta interface{}) (b bool, e error) {
```

Comment:
Since you're  doing the same call to GetIntegrationPD() as in Read, I don't think you need to implement Exists here.